pkg/services/quota/quota.go

```diff
@@ -118,40 +128,107 @@ func (qs *QuotaService) getQuotaScopes(target string) ([]models.QuotaScope, erro
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Global.Org,
+			},
+			models.QuotaScope{
+				Name:             "user",
+				Target:           "org_user",
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.User.Org,
+			},
 		)
 		return scopes, nil
 	case "dashboard":
 		scopes = append(scopes,
-			models.QuotaScope{Name: "global", Target: target, DefaultLimit: qs.Cfg.Quota.Global.Dashboard},
-			models.QuotaScope{Name: "org", Target: target, DefaultLimit: qs.Cfg.Quota.Org.Dashboard},
+			models.QuotaScope{
+				Name:             "global",
+				Target:           target,
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Global.Dashboard,
+			},
+			models.QuotaScope{
+				Name:             "org",
+				Target:           target,
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Org.Dashboard,
+			},
 		)
 		return scopes, nil
 	case "data_source":
 		scopes = append(scopes,
-			models.QuotaScope{Name: "global", Target: target, DefaultLimit: qs.Cfg.Quota.Global.DataSource},
-			models.QuotaScope{Name: "org", Target: target, DefaultLimit: qs.Cfg.Quota.Org.DataSource},
+			models.QuotaScope{
+				Name:             "global",
+				Target:           target,
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Global.DataSource,
+			},
+			models.QuotaScope{
+				Name:             "org",
+				Target:           target,
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Org.DataSource,
+			},
 		)
 		return scopes, nil
 	case "api_key":
 		scopes = append(scopes,
-			models.QuotaScope{Name: "global", Target: target, DefaultLimit: qs.Cfg.Quota.Global.ApiKey},
-			models.QuotaScope{Name: "org", Target: target, DefaultLimit: qs.Cfg.Quota.Org.ApiKey},
+			models.QuotaScope{
+				Name:             "global",
+				Target:           target,
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Global.ApiKey,
+			},
+			models.QuotaScope{
+				Name:             "org",
+				Target:           target,
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Org.ApiKey,
+			},
 		)
 		return scopes, nil
 	case "session":
 		scopes = append(scopes,
-			models.QuotaScope{Name: "global", Target: target, DefaultLimit: qs.Cfg.Quota.Global.Session},
+			models.QuotaScope{
+				Name:             "global",
+				Target:           target,
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Global.Session,
+			},
 		)
 		return scopes, nil
 	default:
 		return scopes, ErrInvalidQuotaTarget
 	}
 }
+
+// get any extra conditions that are required to properly
+// determine the quota for a given target
+func (qs *QuotaService) getQuotaScopeConditions(target string) string {
```

Comment:
The reason I chose here for the function was that I preferred it be passed up the chain to the Query. Additionally this centralizes the conditions based on the target, rather than having random SQL conditions scattered around various models (well, potentially scattered, since this may be the only occasion, or one of few, where additional conditions are required in order to identify the correct quota).

Reply:
I'm also open to the name, since it's conditions related to the target of the quota scope rather than the quota scope itself.  I considered `getQuotaScopeConditionsByTarget` but I felt like that may be redundant, and isn't done above (i.e. it is `QuotaReached` and not `QuotaReachedByTarget`)